_posts/2025-04-09-#31664.md

```diff
@@ -10,17 +10,96 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### Motivation and Background
 
+The current Fee Estimator in Bitcoin Core ([`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/4c1906a500cacab385b09e780b54271b0addaf4b/src/policy/fees.h#L148)) used by the wallet and exposed to clients via RPC, has two key limitations that affect users:
 
-## Questions
+1. **Mempool unaware**: Does not consider the current mempool unconfirmed transactions when estimating fees.
+
+2. **Package unaware**: Only considers individual transactions, ignoring parent-child relationships.
+
+These limitations cause practical problems for:
+- Bitcoin Core wallet users (the majority now rely on third-party providers)
+- Lightning implementations: (e.g., C-lightning)
+- Bitcoin clients: ([e.g., BTCPay Server](https://github.com/btcpayserver/btcpayserver/pull/5490))
+- Others: ([e.g., Decred DEX](https://github.com/decred/dcrdex/pull/3046))
+
+This PR is part of Bitcoin Core project [#30392](https://github.com/bitcoin/bitcoin/issues/30392) which aims to address these issues.
+A detailed design document of the project is available: [Fee Rate Forecasting Design](https://gist.github.com/ismaelsadeeq/bcfa6b1216eef10076665551d9885722)
+
+### Design Goals of `#31664`
+
+1. Creates a pluggable architecture for multiple fee estimation strategies
+2. Create the pathway for making fee estimation in Bitcoin Core both mempool-aware and package-aware
+3. Maintains backward compatibility with existing code
+
+```
+                           ┌─────────────────┐
+                           │FeeRateForecaster│
+                           │    Manager      │
+                           └───────┬─────────┘
+                                   │ owns
+                     ┌─────────────┴───────────────┐
+                     │                             │
+            ┌────────▼─────────┐         ┌─────────▼────────┐
+            │CBlockPolicy      │         │Mempool           │
+            │Estimator         │         │Forecaster        │
+            └──────────────────┘         └──────────────────┘
+            (historical mempool data)    (current mempool)
+```
+
+
+### Implementation Details
+
+PR `#31664` introduces these key components:
```

Comment:
General grammar/style nit: you use a mixture of structures here and in the other bulleted/numbered lists. Sometimes things are complete sentences and sometimes they are fragments, and in a mixture of tenses, e.g. we have "renamed" and "creates" and "create".

Also, I think we are missing some notes on how these components relate to each other?

Reply:
Done and add more notes on the components relationship